specification/search/resource-manager/Microsoft.Search/preview/2019-10-01-preview/search.json

```diff
@@ -1106,6 +1423,190 @@
+          "description": "Describes the properties of a Shared Private Link Resource managed by the Azure Cognitive Search service."
+        }
+      },
+      "description": "Describes a Shared Private Link Resource managed by the Azure Cognitive Search service."
+    },
+    "SharedPrivateLinkResourceProperties": {
+      "properties": {
+        "privateLinkResourceId": {
+          "type": "string",
+          "description": "The resource id of the resource the shared private link resource is for."
+        },
+        "groupId": {
+          "type": "string",
+          "description": "The group id from the provider of resource the shared private link resource is for."
+        },
+        "requestMessage": {
+          "type": "string",
+          "description": "The request message for requesting approval of the shared private link resource."
+        },
+        "status": {
+          "type": "string",
+          "description": "Status of the shared private link resource. Can be Pending, Approved, Rejected, Disconnected, or Timeout.",
+          "enum": [
+            "Pending",
+            "Approved",
+            "Rejected",
+            "Disconnected",
+            "Timeout"
+          ],
+          "x-ms-enum": {
+            "name": "SharedPrivateLinkResourceStatus",
+            "modelAsString": false
+          }
+        }
+      },
+      "description": "Describes the properties of an existing Shared Private Link Resource managed by the Azure Cognitive Search service."
+    },
+    "SharedPrivateLinkResourceListResult": {
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SharedPrivateLinkResource"
+          },
+          "description": "The list of Shared Private Link Resources."
+        }
+      },
+      "description": "Response containing a list of Shared Private Link Resources.",
+      "x-ms-external": true
+    },
+    "PrivateLinkResourcesResult": {
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateLinkResource"
+          },
+          "description": "The list of supported Private Link Resources."
+        }
+      },
+      "description": "Response containing a list of supported Private Link Resources.",
+      "x-ms-external": true
+    },
+    "PrivateLinkResource": {
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The ID of the private link resource."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The name of the private link resource."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        },
+        "properties": {
```

Comment:
You should reference the common types to bring in 'id', 'name', 'type'...

[https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v1/types.json](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v1/types.json)


it has definitions for TrackedResource and ProxyResource

Reply:
Hmmm, interesting.  It seems like we've written our own version of the Resource type, which is very similar to your definition for TrackedResource.  In theory we could also use the ProxyResource for this property.  However, I'm going to leave it as is for now and not risk any major breaking changes that could come with rewriting the swagger this way.  I'll create a work item for the team that properly owns this on our side to address it later (I'm just helping out with Jedi requirements since its been so crazy myself, so I don't think its my place to rewrite this using those definitions).

Reply:
Work item # 641623 in our  back log.